Add types to logging.py

InvincibleRMC opened the pull request:
I was running mypy on my repository and it was complaining about get_logger() not  being typed. So I added static typing to logging.py. Look at previous issues and pull requests.  This change was already done similarly in #979 in response to issue #976. @clalancette mentioned that it should be split up into smaller pull requests.

sloretz commented:
CI ([repos file](https://gist.githubusercontent.com/sloretz/4c5b53ccde1256e3b0a93ff527fe5fd3/raw/a033a5e2ee84fed561235a24b2ed342093e8001f/ros2.repos) build: `--packages-up-to rclpy` test: `--packages-select rclpy`)
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=20368)](http://ci.ros2.org/job/ci_linux/20368/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=14776)](http://ci.ros2.org/job/ci_linux-aarch64/14776/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=21118)](http://ci.ros2.org/job/ci_windows/21118/)

sloretz commented on rclpy/rclpy/logging.py:
```diff
@@ -43,22 +43,22 @@ def clear_config():
 
 
-def set_logger_level(name, level, detailed_error=False):
+def set_logger_level(name: str, level: int | LoggingSeverity, detailed_error: bool = False):
```
Looks like CI is unhappy due to Windows being on Python 3.8. Mind using `Union[int, LoggingSeverity]` instead?

```
    def set_logger_level(name: str, level: int | LoggingSeverity, detailed_error: bool = False):
E   TypeError: unsupported operand type(s) for |: 'type' and 'EnumMeta'
```

InvincibleRMC commented:
I removed the return from the initialize, shutdown, set_logger_level functions. It seems to me that pybinded c++ functions return void. I also added other return types.

fujitatomoya reviewed (approved):
lgtm with green CI

fujitatomoya commented:
> CI ([repos file](https://gist.githubusercontent.com/sloretz/4c5b53ccde1256e3b0a93ff527fe5fd3/raw/a033a5e2ee84fed561235a24b2ed342093e8001f/ros2.repos) build: `--packages-up-to rclpy` test: `--packages-select rclpy`)
> 
> * Linux [![Build Status](https://camo.githubusercontent.com/cf48af2bdadbd4cc206c74e3d5800589e20096215da4b1e96a3bb9d596be1fc1/687474703a2f2f63692e726f73322e6f72672f6275696c645374617475732f69636f6e3f6a6f623d63695f6c696e7578266275696c643d3230333638)](http://ci.ros2.org/job/ci_linux/20368/)
> * Linux-aarch64 [![Build Status](https://camo.githubusercontent.com/ded4831691070eca98278e831f028716a7a10fd6c8ef973dc8655aca171eeb49/687474703a2f2f63692e726f73322e6f72672f6275696c645374617475732f69636f6e3f6a6f623d63695f6c696e75782d61617263683634266275696c643d3134373736)](http://ci.ros2.org/job/ci_linux-aarch64/14776/)
> * Windows [![Build Status](https://camo.githubusercontent.com/1f54dd358362b3f7c221ac2182cb01981c34474eb037d50ac05001aa7122dbe2/687474703a2f2f63692e726f73322e6f72672f6275696c645374617475732f69636f6e3f6a6f623d63695f77696e646f7773266275696c643d3231313138)](http://ci.ros2.org/job/ci_windows/21118/)

CI(retry):
* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=20371)](http://ci.ros2.org/job/ci_linux/20371/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=14780)](http://ci.ros2.org/job/ci_linux-aarch64/14780/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=21121)](http://ci.ros2.org/job/ci_windows/21121/)
